public/content/developers/docs/data-availability/blockchain-data-storage-strategies/index.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Blockchain Data Storage Strategies
+description: There are several ways to store data using the blockchain. This article will compare the different strategies, their costs and tradeoffs, as well as the requirements to use it safely.
+lang: en
+---
+
+There are multiple ways to store information either directly on the blockchain, or in a manner that is secured by the blockchain:
+
+- EIP-4844 blobs
+- Calldata
+- Offchain with L1 mechanisms
+- Contract "code"
+- Events
+- EVM storage
+
+The choice of which method to use is based on several criteria:
+
+- The source of the information. Information in calldata cannot come from the blockchain itself.
+- The destination of the information. Calldata is available only in the transaction it initiates. Events are not accessible onchain.
+- How much hassle is acceptable? Computers that run a full-scale node can perform more processing than a light client in an application running in a browser.
+- Is it necessary to facilitate easy access to the information from every node?
+
+## Prerequisites {#prerequisites}
+
+You should have a good understanding of [blockchain fundamentals](/developers/docs/intro-to-ethereum/). This page also assumes the reader is familiar with [blocks](/developers/docs/blocks/), [transactions](/developers/docs/transactions/), and other relevant topics.
+
+## The C.I.A. of information security {#cia-info-security}
+
+There are three attributes we'd like to have for data.
+
+- _Confidentiality_, unauthorized entities are not allowed to read the information. This is useful in many cases, but _there are no secrets on the blockchain_. Blockchains work because everybody can verify the state transitions, so it is impossible to use them to store secrets directly. There are ways to store confidential information on the blockchain, but they all rely on some offchain component to store at least a key.
+
+- _Integrity_, the information is correct, it cannot be changed by unauthorized entities, or in unauthorized ways (for example, transferring [ERC-20 tokens](https://eips.ethereum.org/EIPS/eip-20#events) without a `Transfer` event). On the blockchain, every node verifies every state change, which ensures integrity.
+
+- _Availability_, the information is available to any authorized entity. On the blockchain, this is usually achieved by having the information available on every [full node](https://ethereum.org/developers/docs/nodes-and-clients#full-node).
```

Comment:
@qbzzt I like this section, but it feels awkwardly placed... how does this place into the rest of the content? Maybe we add a little more "intro" text to tell the user what to expect on this page, and maybe set the stage for why we're showing the CIA section here.

Reply:
I moved and modified, and I think it makes more sense now.